### PR TITLE
fix c style casting

### LINKED ISSUE #2558: Remove c-style pointer casts
https://www.codacy.com/app/Cockatrice/Cockatrice/issues?bid=4363391&filters=W3siaWQiOiJMYW5ndWFnZSIsInZhbHVlcyI6WyJDUFAiXX0seyJpZCI6IkNhdGVnb3J5IiwidmFsdWVzIjpbIkNvZGUgU3R5bGUiXX0seyJpZCI6IkxldmVsIiwidmFsdWVzIjpbbnVsbF19LHsiaWQiOiJQYXR0ZXJuIiwidmFsdWVzIjpbMjEzMl19LHsiaWQiOiJBdXRob3IiLCJ2YWx1ZXMiOltudWxsXX0seyJ2YWx1ZXMiOltdfV0=

See https://www.securecoding.cert.org/confluence/display/cplusplus/EXP05-CPP.+Do+not+use+C-style+casts

### PR DESCRIPTION
**Pull request by ZeldaZach**
## Related Ticket(s)
- Fixes #2558 

## Short roundup of the initial problem
Changes c style casting to C++ style

### CODE REVIEW

File: cockatrice/src/qt-json/json.cpp [hunk 1]
```diff
@@ -269,8 +269,7 @@ QVariant Json::parseObject(const QString &json, int &index, bool &success)
         Json::nextToken(json, index);
 
         //Loop through all of the key/value pairs of the object
-        bool done = false;
-        while(!done)
+        while(true)
```
Comment (Daenyth) [inline cockatrice/src/qt-json/json.cpp]: Did you mean to do this?

File: cockatrice/src/filtertree.cpp [hunk 2]
```diff
@@ -27,7 +27,7 @@ int FilterTreeBranch<T>::childIndex(const FilterTreeNode *node) const
 
     /* to do the dynamic cast to T we will lose const'ness, but we can
```
Comment (Daenyth) [inline cockatrice/src/filtertree.cpp]: I think the comment can go away now that we are using `const_cast`

File: cockatrice/src/qt-json/json.cpp (hunk 1 above)
Comment (ZeldaZach) [inline cockatrice/src/qt-json/json.cpp]: I changed it intentionally. The review site said that done was never changed to true anywhere in the code and I looked so I just changed it to true to shut them up

Comment (Nightfirecat) [inline cockatrice/src/qt-json/json.cpp]: Maybe the solution would be to use a function rather than an infinite loop to get this return value?

Comment (Daenyth) [inline cockatrice/src/qt-json/json.cpp]: Let's not change it. We can tell codacy to ignore the thing. This is code we vendored in - I can actually set up an ignore on the whole directory

**Pull request merged**